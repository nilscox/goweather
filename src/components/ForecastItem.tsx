/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import { Card, CardText, CardBody, CardTitle, Container, Row, Col } from 'reactstrap';

import { IWeather } from '../interfaces';

const ForecastItem = ({ date, description, temperature, humidity, pressure }: IWeather) => (
  <Card css={cardStyle} className="my-2">
    <CardBody>

      <CardTitle>{ date.format('dddd, MMMM Do') }</CardTitle>

      <CardText className="text-center">
        <strong>{ description }</strong>
      </CardText>

      <Container>
        <Row>
          <Col xs="12" sm="4" className="text-center">Temperature: { temperature }Â°C</Col>
          <Col xs="12" sm="4" className="text-center">Humidity: { humidity }%</Col>
          <Col xs="12" sm="4" className="text-center">Pressure: { pressure }hPa</Col>
        </Row>
      </Container>

    </CardBody>
  </Card>
);

const cardStyle = css`

`;

export default ForecastItem;
