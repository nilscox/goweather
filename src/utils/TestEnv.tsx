import * as React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';

import store from '../store';

type TestEnvProps = {
  children: React.ReactNode;
};

const TestEnv = ({ children }: TestEnvProps) => (
  <Provider store={store}>
    <Router>
      { children }
    </Router>
  </Provider>
);

export default TestEnv;
