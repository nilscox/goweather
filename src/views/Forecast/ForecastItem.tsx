/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Card, CardText, CardBody, CardTitle, Container, Row, Col } from 'reactstrap';

import { IWeather } from '../../interfaces';

type ForecastItemProps = IWeather;

/**
 * Display a day of forecast weather.
 * TODO: add an image (or icon)
 */
const ForecastItem = ({ date, description, temperature, humidity, pressure }: ForecastItemProps) => (
  <Card className="my-2">
    <CardBody>

      <CardTitle>{ date.format('dddd, MMMM Do') }</CardTitle>

      <CardText className="text-center">
        <strong>{ description }</strong>
      </CardText>

      <Container>
        <Row>
          <Col xs="12" sm="4" className="text-center">Temperature: { temperature }Â°C</Col>
          <Col xs="12" sm="4" className="text-center">Humidity: { humidity }%</Col>
          <Col xs="12" sm="4" className="text-center">Pressure: { pressure }hPa</Col>
        </Row>
      </Container>

    </CardBody>
  </Card>
);

export default ForecastItem;
