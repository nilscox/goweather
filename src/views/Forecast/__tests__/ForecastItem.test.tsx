import * as React from 'react';
import * as renderer from 'react-test-renderer';
import * as moment from 'moment';

import TestEnv from '../../../utils/TestEnv';

import ForecastItem from '../ForecastItem';

test('ForecastItem', () => {
  const tree = renderer
    .create(
      <TestEnv>
        <ForecastItem
          date={moment('2018-04-06 13:42')}
          description="light rain"
          humidity={100}
          pressure={1037.7}
          temperature={1.3}
        />
      </TestEnv>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
