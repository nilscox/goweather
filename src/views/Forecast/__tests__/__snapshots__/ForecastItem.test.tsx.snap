// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem 1`] = `
<div
  className="my-2 card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Friday, April 6th
    </h5>
    <p
      className="text-center card-text"
    >
      <strong>
        light rain
      </strong>
    </p>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="text-center col-12 col-sm-4"
        >
          Temperature: 
          1.3
          Â°C
        </div>
        <div
          className="text-center col-12 col-sm-4"
        >
          Humidity: 
          100
          %
        </div>
        <div
          className="text-center col-12 col-sm-4"
        >
          Pressure: 
          1037.7
          hPa
        </div>
      </div>
    </div>
  </div>
</div>
`;
